src/email_generator_logic/email_generator.py

```diff
@@ -0,0 +1,72 @@
+import requests
+import email_endpoints
+
+class OneSecMail():
+    def __init__(self):
+        self.session = requests.session()
+        self.mails = []
+        self.domains = []
+        self.usernames = []
+        self.messages = []
+        self.lastMessages = []
+        self.lastMessagesBody = []
+        self.lastMessagesAttachment = []
+        self.lastMessagesAttachmentName = []
+
+    def create_new_emails(self, numberOfEmails):
+        self.mails = self.session.get(email_endpoints.NEW_EMAILS + str(numberOfEmails)).json()        
+        for email in self.mails:
+            self.messages.append([])
+            self.lastMessagesAttachment.append("")
+            self.lastMessagesAttachmentName.append("")
+            self.lastMessages.append("")
+            self.lastMessagesBody.append("")
+            self.usernames.append(email.split("@")[0])
+            self.domains.append(email.split("@")[1])
+
+    def update_messages(self):
+        for idx in range(len(self.mails)):
+            self.messages[idx] = self.session.get(email_endpoints.CHECK_INBOXES.format(self.usernames[idx], self.domains[idx])).json()
+
+    def update_last_messages(self):
+        for idx in range(len(self.mails)):
+            try:
+                self.lastMessages[idx] = self.session.get(email_endpoints.CHECK_EXACT_MESSAGES.format(self.usernames[idx], self.domains[idx], self.messages[idx][-1]['id'])).json()
+                self.lastMessagesBody[idx]= self.lastMessages[idx]['textBody']
+                try:
+                    self.lastMessagesAttachmentName[idx]= self.lastMessages[idx]['attachments']['filename']
+                except:
+                    pass
```

Comment:
```suggestion
  logger.info("reading attachment from mail failed")
```
Warto dodać logowanie w takich przypadkach bo potem łatwo gubić błędy

Reply:
pewnie tylko to passuje każdą co nie dostała więc albo wszystkie dostaną albo żadne więc to jest na potrzeby testów tak naprawdę.